File: src/rosterizer/Associate.java
package rosterizer;

public class Associate {

    public enum Role {
        AMBASSADOR(1),
        EOL(2),
        PS(4),
        WATERSPIDER(8),
        REFURB(16),
        WHD(32),
        RECOVERY(64),
        OUTBOUND(128),
        PROCESS(256),
        ICQA(512),
        TDR(1024),
        PIT(2048),
        SHOES(4096);

        final int value;

        Role(int value) {
            this.value = value;
        }
    }

    private int roles = 0;
    private final String username;

    public Associate(String username) {
        this.username = username;
    }

    public void addTrainedRole(int newRole) {
        roles = roles | newRole;
    }

    public boolean hasRole(int role) {
        return (roles & role) == role;
    }

    public String getUsername() { return username; }

    @Override
    public String toString() {
        return String.format("%s,%d", username, roles);
    }
}

File: src/rosterizer/Associate.kt
package rosterizer

class Associate(val login: String) {
    companion object {
        const val ROLE_AMBASSADOR = 1
        const val ROLE_EOL = 2
        const val ROLE_PS = 4
        const val ROLE_WATERSPIDER = 8
        const val ROLE_REFURB = 16
        const val ROLE_WHD = 32
        const val ROLE_RECOVERY = 64
        const val ROLE_OUTBOUND = 128
        const val ROLE_SHOES = 256
        const val ROLE_ICQA = 512
        const val ROLE_TDR = 1024
        const val ROLE_PIT = 2048
    }

    private var roles = 0
    private var rate = 0.0
    private var inferredTime = 0.0

    fun addTrainedRole(newRole: Int) { roles = roles or newRole }
    fun hasRole(role: Int): Boolean = (roles and role) == role

    override fun toString(): String {
        return "$login {$roles}"
    }
}